src/core/src/pool.rs

```diff
@@ -14,90 +14,80 @@
+impl<B: Backend> GeneralCommandPool<B> {
     /// Get a command buffer for recording.
     ///
     /// You can only record to one command buffer per pool at the same time.
     /// If more command buffers are requested than allocated, new buffers will be reserved.
     /// The command buffer will be returned in 'recording' state.
-    fn acquire_command_buffer<'a>(
-        &'a mut self,
-    ) -> command::Encoder<'a, B, GraphicsCommandBuffer<B>>;
+    pub fn acquire_command_buffer(&mut self) -> Encoder<B, GeneralCommandBuffer<B>> {
+        unsafe { Encoder::new(GeneralCommandBuffer(self.0.acquire_command_buffer())) }
+    }
 }
-
-/// Compute command pool can allocate compute command buffers.
-pub trait ComputeCommandPool<B: Backend>: CommandPool<B> {
-    ///
-    fn from_queue<Q>(queue: Q, capacity: usize) -> Self
-    where
-        Q: Compatible<ComputeQueue<B>> + AsRef<B::CommandQueue>;
-
+///
+pub struct GraphicsCommandPool<B: Backend>(pub(crate) B::RawCommandPool);
+impl<B: Backend> GraphicsCommandPool<B> {
     /// Get a command buffer for recording.
     ///
     /// You can only record to one command buffer per pool at the same time.
     /// If more command buffers are requested than allocated, new buffers will be reserved.
     /// The command buffer will be returned in 'recording' state.
-    fn acquire_command_buffer<'a>(&'a mut self)
-        -> command::Encoder<'a, B, ComputeCommandBuffer<B>>;
+    pub fn acquire_command_buffer(&mut self) -> Encoder<B, GraphicsCommandBuffer<B>> {
+        unsafe { Encoder::new(GraphicsCommandBuffer(self.0.acquire_command_buffer())) }
+    }
 }
-
-/// Transfer command pool can allocate transfer command buffers.
-pub trait TransferCommandPool<B: Backend>: CommandPool<B> {
-    ///
-    fn from_queue<Q>(queue: Q, capacity: usize) -> Self
-    where
-        Q: Compatible<TransferQueue<B>> + AsRef<B::CommandQueue>;
-
+///
+pub struct ComputeCommandPool<B: Backend>(pub(crate) B::RawCommandPool);
+impl<B: Backend> ComputeCommandPool<B> {
     /// Get a command buffer for recording.
     ///
     /// You can only record to one command buffer per pool at the same time.
     /// If more command buffers are requested than allocated, new buffers will be reserved.
     /// The command buffer will be returned in 'recording' state.
-    fn acquire_command_buffer<'a>(
-        &'a mut self,
-    ) -> command::Encoder<'a, B, TransferCommandBuffer<B>>;
+    pub fn acquire_command_buffer(&mut self) -> Encoder<B, ComputeCommandBuffer<B>> {
+        unsafe { Encoder::new(ComputeCommandBuffer(self.0.acquire_command_buffer())) }
+    }
 }
-
-/// Subpass command pool can allocate subpass command buffers.
-pub trait SubpassCommandPool<B: Backend>: CommandPool<B> {
-    ///
-    fn from_queue<Q>(queue: Q, capacity: usize) -> Self
-    where
-        Q: Compatible<GraphicsQueue<B>> + AsRef<B::CommandQueue>;
-
+///
+pub struct TransferCommandPool<B: Backend>(pub(crate) B::RawCommandPool);
```

Comment:
oh, aren't you quick to use the new shiny features? 😁  I don't mind at all, just need to get used to it as well